dali/pipeline/executor/executor.cc

```diff
@@ -243,6 +243,44 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunGPU() {
   QueuePolicy::QueueOutputIdxs(gpu_idxs, gpu_op_stream_);
 }
 
+template <typename WorkspacePolicy, typename QueuePolicy>
+void Executor<WorkspacePolicy, QueuePolicy>::RunCPU() {
+  try {
+    RunCPUImpl();
+  } catch (std::exception &e) {
+    HandleError(make_string("Exception in CPU stage: ", e.what()));
+    throw;
```

Comment:
....should we rethrow? Perhaps only in synchronous executor?

Reply:
No, we shouldn't. The whole point is executor collects the errors and we throw them from main thread, as otherwise it kills Python (or hangs).